--- json_save_cc.cc	2020-02-23 13:20:55.283898093 +0000
+++ json_save_cc.cc	2020-02-23 13:20:30.051086971 +0000
@@ -1,6 +1,6 @@
 #include <octave/oct.h>
 
-#if OCTAVE_MAJOR_VERSION >= 4 and OCTAVE_MINOR_VERSION >= 4
+#if OCTAVE_MAJOR_VERSION >= 5 or (OCTAVE_MAJOR_VERSION >= 4 and OCTAVE_MINOR_VERSION >= 4)
 #  include <octave/interpreter.h>
 #  define OCT_STREAM_TYPE    octave::stream
 #  define OCT_REGEXP_REPLACE octave::regexp::replace
