--- client.py	2020-01-24 21:05:52.530626977 +0000
+++ client.py	2020-02-03 20:51:42.290914515 +0000
@@ -292,12 +292,12 @@
         else:
             logging.debug("recv MSG (%s) %s: %s", self._stream_name, name, value)
         # Handle error conditions
-        if name == 'too_busy':
-            raise KiwiTooBusyError('%s: all %s client slots taken' % (self._options.server_host, value))
-        if name == 'badp' and value == '1':
-            raise KiwiBadPasswordError('%s: bad password' % self._options.server_host)
-        if name == 'down':
-            raise KiwiDownError('%s: server is down atm' % self._options.server_host)
+        # if name == 'too_busy':
+        #     raise KiwiTooBusyError('%s: all %s client slots taken' % (self._options.server_host, value))
+        # if name == 'badp' and value == '1':
+        #     raise KiwiBadPasswordError('%s: bad password' % self._options.server_host)
+        # if name == 'down':
+        #     raise KiwiDownError('%s: server is down atm' % self._options.server_host)
         # Handle data items
         if name == 'audio_rate':
             self._set_ar_ok(int(value), 44100)
