--- tdoa_plot_map.m	2018-08-09 08:58:20.544344593 +0000
+++ tdoa_plot_map.m	2018-08-09 09:00:16.825180472 +0000
@@ -73,7 +73,7 @@
   ha = axes('Position',[0 0 1 1],'Xlim',[0 1],'Ylim',[0  1],'Box','off','Visible','off','Units','normalized', 'clipping' , 'off');
   text(0.5, 0.98,  plot_info.title, 'fontweight', 'bold', 'horizontalalignment', 'center', 'fontsize', 15);
 
-  print('-dpng','-S900,600', fullfile('png', sprintf('%s.png', plot_info.plotname)));
+##  print('-dpng','-S900,600', fullfile('png', sprintf('%s.png', plot_info.plotname))); nopng patch for directTDoA v2.6x
   print('-dpdf','-S900,600', fullfile('pdf', sprintf('%s.pdf', plot_info.plotname)));
 endfunction
 
