--- tdoa_plot_map.m	2020-02-03 21:02:16.088313158 +0000
+++ tdoa_plot_map.m	2020-02-02 00:06:29.207199473 +0000
@@ -140,6 +140,7 @@
     end
     set(colorbar(),'XLabel', 'sqrt(\chi^2)/ndf')
     printf('tdoa_plot_map_combined: [%.3f sec]\n', toc());
+    global mlp
     ha = axes('Position', [0 0 1 1], ...
               'Xlim',     [0 1], ...
               'Ylim',     [0 1], ...
@@ -147,11 +148,11 @@
               'Visible',  'off', ...
               'Units',    'normalized', ...
               'clipping', 'off');
-    text(0.5, 0.98,  plot_info.title, ...
+    text(0.5, 0.98,  strcat(plot_info.title, mlp), ...
          'fontweight', 'bold', ...
          'horizontalalignment', 'center', ...
          'fontsize', 15);
-    print('-dpng','-S900,600', fullfile('png', sprintf('%s.png', plot_info.plotname)));
+    # print('-dpng','-S900,600', fullfile('png', sprintf('%s.png', plot_info.plotname)));
     print('-dpdf','-S900,600', fullfile('pdf', sprintf('%s.pdf', plot_info.plotname)));
   end
   if plot_info.plot_kiwi_json
@@ -184,6 +185,7 @@
   _i = _i(_j);
   pos = [plot_info.lat(_i) plot_info.lon(_j)];
   printf('most likely position: lat = %.2f deg  lon = %.2f deg\n', pos);
+  global mlp=cstrcat(sprintf(' - [lat %.2f lon %.2f]', plot_info.lat(_i), plot_info.lon(_j)));
 endfunction
 
 function [bb_lon, bb_lat, idx_lon, idx_lat] = find_bounding_box(plot_info, h)
